src/FsAutoComplete/CodeFixes/IgnoreExpression.fs

```diff
@@ -0,0 +1,58 @@
+module FsAutoComplete.CodeFix.IgnoreExpression
+
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Text
+open FsToolkit.ErrorHandling
+open Ionide.LanguageServerProtocol.Types
+open FsAutoComplete.CodeFix.Types
+open FsAutoComplete
+open FsAutoComplete.LspHelpers
+
+let title = "Ignore expression"
+
+let fix (getParseResultsForFile: GetParseResultsForFile) : CodeFix =
+  Run.ifDiagnosticByCode (set [ "20" ]) (fun diagnostic (codeActionParams: CodeActionParams) ->
+    asyncResult {
+      let fileName = codeActionParams.TextDocument.GetFilePath() |> Utils.normalizePath
+      let fcsPos = protocolPosToPos codeActionParams.Range.Start
+
+      let! (parseAndCheckResults: ParseAndCheckResults, _line: string, sourceText: IFSACSourceText) =
+        getParseResultsForFile fileName fcsPos
+
+      let mDiag =
+        protocolRangeToRange parseAndCheckResults.GetParseResults.FileName diagnostic.Range
+
+      let mExprOpt =
+        (fcsPos, parseAndCheckResults.GetParseResults.ParseTree)
+        ||> ParsedInput.tryPick (fun path node ->
+          match node with
+          | SyntaxNode.SynExpr(e) when Range.equals mDiag e.Range -> Some(path, e)
+          | _ -> None)
+
+      match mExprOpt with
+      | None -> return []
+      | Some(path, expr) ->
+        // Only do single line for now
+        if expr.Range.StartLine <> expr.Range.EndLine then
+          return []
+        else
+          let needsParentheses =
+            SynExpr.shouldBeParenthesizedInContext sourceText.GetLineString path expr
```

Comment:
Should this be checking whether parens are needed once the original expression becomes the left-hand argument to the infix `|>` application?

Something like

```fsharp
let needsParentheses =
  let exprInPipe =
    SynExpr.App(
      ExprAtomicFlag.NonAtomic,
      false,
      SynExpr.App(
        ExprAtomicFlag.NonAtomic,
        true,
        SynExpr.LongIdent(false, SynLongIdent([Ident("op_PipeRight", Range.range0)], [], [Some(IdentTrivia.OriginalNotation "|>")]), None, Range.range0),
        expr, // The expr that will now be piped into ignore.
        Range.range0
      ),
      SynExpr.Ident(Ident("ignore", Range.range0)),
      Range.range0 // Range shouldn't matter in this case, I think.
    )

  SynExpr.shouldBeParenthesizedInContext sourceText.GetLineString path exprInPipe

Reply:
Yes, it should, good catch!

Reply:
As an aside: if certain hypotheticals like this became commonplace enough, maybe it would make sense to add helpers like `SynExpr.mkInfixApp left op right`, etc., to FCS. Or even just APIs for easy round-tripping (AST → source text → AST) of source text fragments?

Reply:
I think it might make sense to have such a mechanism in FSAC first.
Depends on future use-cases, but yes I do see some abstractions pop up at some point.